File: backend/core-service/src/Fix.Application/UseCases/Mandates/Commands/MandateTermsInput.cs
using Fix.Domain.Common;

namespace Fix.Application.Mandates.Commands;

/// <summary>Termos do mandato informados no contrato (emissão, edição e prévia de enquadramento).</summary>
public sealed record MandateTermsInput(
    string Title,
    string? Criteria,
    Commodity? Commodity,
    string? Tenor,
    decimal? Quantity,
    MeasurementUnit? QuantityUnit,
    bool AtMarket,
    decimal? PriceTarget,
    decimal? PriceMin,
    decimal? PriceMax,
    string? PriceUnit,
    DateOnly? WindowStart,
    DateOnly? WindowEnd);

File: frontend/bff/src/application/mandates/commands/mandate-terms.input.ts
import type { Commodity } from '../../../cross-cutting/enums/commodity.js';
import type { MeasurementUnit } from '../../../cross-cutting/enums/measurement-unit.js';
import type { PriceCriteriaInput } from './price-criteria.input.js';

export interface MandateTermsInput {
  title: string;
  criteria: string | null;
  commodity: Commodity;
  /** Tela/vencimento: N26, fev/27. */
  tenor: string | null;
  quantity: number | null;
  quantityUnit: MeasurementUnit;
  price: PriceCriteriaInput;
  windowStart: string | null;
  windowEnd: string | null;
}
